File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.swift
class Solution {
    func minLength(_ s: String) -> Int {
        var stack = s.map { Int($0.asciiValue!) }
        var result = [Int]()
        var count = 0

        loopStack(stack: stack)

        func loopStack(stack: [Int]) {
            var stacks = stack
            result = stacks
            guard stacks.count != 0 else { return }
            for i in 0 ..< stacks.count - 1 {
                if stacks[i] == 65 && stacks[i + 1] == 66 || stacks[i] == 67 && stacks[i + 1] == 68 {
                    count += 1
                    stacks.remove(at: i + 1)
                    stacks.remove(at: i)
                    loopStack(stack: stacks)
                    break
                }
            }
        }

        if count == 0 {
            return s.count
        } else {
            return result.count
        }
    }
}

File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.kt
class Solution {
    fun minLength(s: String): Int {
        val stack = Stack<Char>()

        for (i in s) {
            if (i == 'B') {
                if (!stack.empty() && stack.peek() == 'A') stack.pop()
                else stack.push(i)
            } else if (i == 'D') {
                if (!stack.empty() && stack.peek() == 'C') stack.pop()
                else stack.push(i)
            } else {
                stack.push(i)
            }
            // println(stack)
        }
        return stack.size
    }
}